core/server/models/base/utils.js

```diff
@@ -94,5 +94,16 @@ detach = function detach(Model, effectedModelId, relation, modelsToAttach, optio
         });
 };
 
+/**
+ * Utility function used by the various model permission hooks
+ * to determine whether or not a user has a particular role
+ */
+actorIs = function actorIs(user, roles) {
```

Comment:
Did you see that we have a `.hasRole` fn in our User model?
I would prefer having both functions in one place.

The permission provider fetches the logged in user including it's relations, the problem is just that it does not return the user instance, only the relations. For now, i would suggest to use a static fn next to the existing instance fn `hasRole`. 

`userModel.hasRole('Owner')`
`models.User.hasRole(user.roles, 'Owner')`